### PR TITLE
Add unit tests for GUID unmigration

### PR DESCRIPTION
**Pull request by nflynt**
## Problem
When we originally merged in the uuid-unmigration, due to active issues, unit test writing was deferred so we could get the fix out more quickly. These are those tests.

To facilitate easier testing, several small refactors were performed:
- The various "identify" functions now accept the lists of objects to operate on, rather than the external state necessary to query those objects
- For LDAP lookups, the retryable connection is hoisted into an interface so it can be more easily mocked

## Testing
These unit tests cover the various utility functions and identification routines. For this initial round I've focused on tests  that doesn't need to touch external services, excepting LDAP lookups, since this minimizes the amount of required mocks. Most of the interesting decisions are made during identification anyway.

### Automated Testing
* Test types added/modified:
    * Unit

## QA Testing Considerations
For the most part, if drone passes then the tests also passed. No manual testing should be required.

### CODE REVIEW

**Comment by nflynt** [comment]
Note: Some review was previously performed in these smaller PRs:
https://github.com/rancher/rancher/pull/42929
https://github.com/rancher/rancher/pull/42911
https://github.com/rancher/rancher/pull/42881

**Review by KevinJoiner — COMMENTED** [review]
There are a lot of refactors here, which is okay, but could you add in the PR template what refactors you made and why, since it is hard to follow just by looking at the code without context?
